pkg/operator/statefulset.go

```diff
@@ -75,13 +75,18 @@ func CreateConfigReloader(
 		Limits:   v1.ResourceList{},
 		Requests: v1.ResourceList{},
 	}
-	if config.CPU != "0" {
-		resources.Limits[v1.ResourceCPU] = resource.MustParse(config.CPU)
-		resources.Requests[v1.ResourceCPU] = resource.MustParse(config.CPU)
+
+	if config.CPURequest != "0" {
+		resources.Requests[v1.ResourceCPU] = resource.MustParse(config.CPURequest)
```

Comment:
Side note for a following PR: we should validate the values earlier in the process rather than panic'ing so late :)

Reply:
I'm not sure what you mean by panicking. There is no zero/nil/null value for resources. The only way to for a container to have no CPU/memory requests/limits is to not set them. These checks are how we accomplish that.

Reply:
I mean that [`resource.MustParse()`](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#MustParse) will panic if the provided value isn't a valid quantity.

Reply:
That makes sense. Want me to add to init() some code to do the check?

It would look like this

```go
	// ...
	flagset.StringVar(&cfg.PromSelector, "prometheus-instance-selector", "", "Label selector to filter Prometheus Custom Resources to watch.")
	flagset.StringVar(&cfg.AlertManagerSelector, "alertmanager-instance-selector", "", "Label selector to filter AlertManager Custom Resources to watch.")
	flagset.StringVar(&cfg.ThanosRulerSelector, "thanos-ruler-instance-selector", "", "Label selector to filter ThanosRuler Custom Resources to watch.")
	flagset.StringVar(&cfg.SecretListWatchSelector, "secret-field-selector", "", "Field selector to filter Secrets to watch")

	// Prioritize specific flags over generic ones
	// If config.CPU has been set but not config.CPURequest/config.CPULimit, default "" to config.CPU
	// If config.Memory has been set but not config.MemoryRequest or config.MemoryLimit, default "" to config.Memory
	if rcCPU != "" {
		if cfg.ReloaderConfig.CPURequest == "" {
			cfg.ReloaderConfig.CPURequest = rcCPU
		}
		if cfg.ReloaderConfig.CPULimit == "" {
			cfg.ReloaderConfig.CPULimit = rcCPU
		}
	}
	if rcMemory != "" {
		if cfg.ReloaderConfig.MemoryRequest == "" {
			cfg.ReloaderConfig.MemoryRequest = rcMemory
		}
		if cfg.ReloaderConfig.MemoryLimit == "" {
			cfg.ReloaderConfig.MemoryLimit = rcMemory
		}
	}
	resource.MustParse(cfg.ReloaderConfig.CPULimit)
	resource.MustParse(cfg.ReloaderConfig.CPURequest)
	resource.MustParse(cfg.ReloaderConfig.MemoryLimit)
	resource.MustParse(cfg.ReloaderConfig.MemoryRequest)
}
```

Reply:
It would be great! Instead of MustParse(), you could call ParseQuantity(), log the error and return from Main().

Reply:
I'll get the change committed.

Reply:
Change has been committed.